src/Vehicle/Vehicle.cc

```diff
@@ -930,7 +930,7 @@ void Vehicle::_handleStatusText(mavlink_message_t& message)
     uint8_t compId = message.compid;
 
     b.resize(MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN+1);
-    strncpy(b.data(), statustext.text, MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN);
+    strcpy(b.data(), statustext.text);
```

Comment:
Rather use `memcpy`, as `statustext.text` might not be null-terminated.
The compiler warning is a false positive in this case, and `strncpy` is correct.

Reply:
> Rather use memcpy, as statustext.text might not be null-terminated.

True, I'll change to use memcpy.

> strncpy is correct

I still don't understand why this is correct. strncpy does compile time length checks right? That is being violated and during runtime we are resizing. Which I believe renders using `strncpy` mute. Not to mention it will remove the null-terminator even if it exists.

Reply:
Just compiled with memcpy and it also gives us the same warning. Maybe we should try to change how `b` is initialized and ensure that it is 50 bytes so we don't have to resize?

Reply:
> I still don't understand why this is correct. strncpy does compile time length checks right? 

It's a runtime-check and gcc has some extra checks at compile-time. Hence the compile-check fails but it's correct at runtime. The null termination is enforced on the next line.

> Just compiled with memcpy and it also gives us the same warning.

I thought it might.

> Maybe we should try to change how b is initialized and ensure that it is 50 bytes so we don't have to resize?

Yes you could directly use this one: https://doc.qt.io/qt-5/qbytearray.html#QByteArray-1

Reply:
@bkueng I have updated the commit with your suggestion.